src/application_settings/container_base.py

```diff
@@ -0,0 +1,267 @@
+                if is_valid_filepath(file_path, platform="auto"):
+                    path = Path(file_path).resolve()
+                else:
+                    raise ValueError(
+                        f"Given path: '{file_path}' is not a valid path for this OS"
+                    )
+
+            if path:
+                _ALL_PATHS[id(cls)] = path
+            else:
+                _ALL_PATHS.pop(id(cls), None)
+
+        @classmethod
+        def filepath(cls) -> Path | None:
+            """Return the path for the file that holds the config / settings."""
+            return _ALL_PATHS.get(id(cls), cls.default_filepath())
+
+        @classmethod
+        def _get(cls: type[_ContainerT]) -> _ContainerT | None:
+            """Private getter for the singleton."""
+            return _ALL_CONTAINERS.get(id(cls))
+
+    else:
+
+        @classmethod
+        def default_filepath(cls: type[_ContainerT]) -> Union[Path, None]:
+            """Return the fully qualified path for the config/settingsfile: e.g. ~/.example/config.toml"""
+            return Path.home() / f"{cls.default_foldername()}" / cls.default_filename()
+
+        @classmethod
+        def set_filepath(
+            cls: type[_ContainerT], file_path: Union[str, Path] = ""
+        ) -> None:
+            """Set the path for the file (a singleton)."""
+
+            path: Union[Path, None] = None
+            if isinstance(file_path, Path):
+                path = file_path.resolve()
+            elif file_path:
+                if is_valid_filepath(file_path, platform="auto"):
+                    path = Path(file_path).resolve()
+                else:
+                    raise ValueError(
+                        f"Given path: '{file_path}' is not a valid path for this OS"
+                    )
+
+            if path:
+                _ALL_PATHS[id(cls)] = path
+            else:
+                _ALL_PATHS.pop(id(cls), None)
+
+        @classmethod
+        def filepath(cls) -> Union[Path, None]:
+            """Return the path for the file that holds the config / settings."""
+            return _ALL_PATHS.get(id(cls), cls.default_filepath())
+
+        @classmethod
+        def _get(cls: type[_ContainerT]) -> Union[_ContainerT, None]:
+            """Private getter for the singleton."""
+            return _ALL_CONTAINERS.get(id(cls))
+
+    @classmethod
+    def get(cls: type[_ContainerT], reload: bool = False) -> _ContainerT:
+        """Access method for the singleton."""
+
+        if (_the_container_or_none := cls._get()) is None or reload:
+            # no config has been made yet or it needs to be reloaded,
+            # so let's instantiate one and keep it in the global store
+            _the_config = cls._create_instance()
+            _the_config._set()
+        else:
+            _the_config = _the_container_or_none
+        return _the_config
+
+    def update(self: _ContainerT, changes: dict[str, dict[str, Any]]) -> _ContainerT:
```

Comment:
Does it not make sense to make this method abstract here, and move this implementation to the `settings_base`. That way some third option that could be created later doesn't accidentally take over this?

Reply:
ahh we can't due to the `_set` and `_save` methods? Maybe it's fine like this as well.

Reply:
Right, because of the `_set` and `_save` methods.